backend/src/middleware/requireBotAuth.ts

```diff
@@ -15,30 +16,22 @@ const requireBotAuth = ({
     location?: req;
 }) => {
     return async (req: Request, res: Response, next: NextFunction) => {
-        try {
-            const bot = await Bot.findOne({ _id: req[location].botId });
-            
-            if (!bot) {
-                throw new Error('Failed to find bot');
-            }
-            
-            await validateMembership({
-                userId: req.user._id.toString(),
-                workspaceId: bot.workspace.toString(),
-                acceptedRoles,
-                acceptedStatuses
-            });
-            
-            req.bot = bot;
-            
-            next();
-        } catch (err) {
-            Sentry.setUser(null);
-			Sentry.captureException(err);
-			return res.status(401).send({
-				error: 'Failed bot authorization'
-			});
+        const bot = await Bot.findOne({ _id: req[location].botId });
+        
+        if (!bot) {
+            return next(UnauthorizedRequestError({message: 'Failed to locate Bot account'}))
```

Comment:
To provide more context on this `Bot` stuff (it's a new feature/mechanism that we've been working on this past 2 weeks) — Every project/workspace has a `Bot` that has its own public-private key pair. The purpose of the bot is for people to be able to share their project/workspace key with it so it can perform various automations including syncing secrets to cloud integrations (can elaborate more on this when we call soon).

This middleware route is basically to check that for whatever `botId` being passed through likely as route params, that it matches to a `Bot` in the database and that `Bot` is a bot of a project that the user on the request is a member of.

Wondering if this would fall under the `UnauthorizedRequestError`? It could be that no `Bot` exists for that `botId` being passed in? I'm cool with leaving it as is but just wanted to provide more context anyways!

Reply:
I wasn't sure to use which error message or should I create a new types of error. So just used `UnauthorizedRequestError` while it was inside of Authentication logic. It's my mistake not to write TODO/FIXME or comment about it. Thanks for giving more context on the topic. Will change to correct erros with next commit. Any suggestions for types, names, or context for this type of error would be appreciated